runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/WindmillStreamShutdownException.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.dataflow.worker.windmill.client;
+
+/** Thrown when operations are requested on a {@link WindmillStream} has been shutdown/closed. */
+public final class WindmillStreamShutdownException extends Exception {
```

Comment:
it's not particularly clear what the difference between StreamClosedException and WIndmillStreamShutdownException are from the comments.

I was wondering too if we need different types of exceptions to distinguish between shutdown and stream being closed when handling them.  IF not we could just have one class with different messages.

Reply:
Renamed StreamClosedException to RetryableStreamClosedException since we will try to flush all of the pending requests in this case.  For shutdown it means that we are no longer operating with the stream as it is invalid

Reply:
I think its also nice to have both since we ignore the StreamClosedException usually and We always propogate the WindmillStreamShutdownException

might be useful to just have send return true/false? instead of throwing StreamClosed?